<?xml version="1.0"?>

<launch>
  <!-- Load joint controller configurations from YAML file to parameter server -->
  <rosparam file    = "$(find vigir_robotiq_ros_control)/config/r_robotiq_states_controller.yaml"
            command = "load"/>
  <rosparam file    = "$(find vigir_robotiq_ros_control)/config/r_robotiq_joint_limits.yaml"
            command = "load"/>
            
  <rosparam ns      = "right_robotiq"
            file    = "$(find vigir_robotiq_ros_control)/config/r_robotiq_position_trajectory_controller.yaml"
            command = "load"/>
            
  <node name    = "right_rosparam_append_node" 
        pkg     = "flor_atlas_moveit_config"
        type    = "rosparam_append"
        args    = "$(find vigir_robotiq_ros_control)/config/r_robotiq_controllers.yaml /move_group/controller_list"/>
            
  <node name    = "right_robotiq_position_trajectory_controller"
        pkg     = "controller_manager"
        type    = "spawner"
        ns      = "right_robotiq"
        output  = "screen"
        args    = "right_hand_traj_controller"/>
        
            
  <!-- load the controllers -->
  <node name    = "right_robotiq_state_controller_spawner"
        pkg     = "controller_manager"
        type    = "spawner" 
        respawn = "false" 
        ns      = "right_robotiq" 
        output  = "screen"  
        args    = "joint_state_controller"/>  
  <node name    = "right_robotiq_ros_controller"
        pkg     = "vigir_robotiq_ros_control" 
        type    = "vigir_robotiq_ros_controller_node" 
        ns      = "right_robotiq"
        output  = "screen">
        <param name="hand_side" value="right" />
        <param name="hand_name" value="r_hand"/>
        <rosparam file    = "$(find vigir_robotiq_ros_control)/config/robotiq_joints.yaml" 
                  command = "load" />
  </node>
  
  <arg name="input_joints"  default="/right_robotiq/joint_states" />  <!-- Relaying to /joint_states since namespace contained hand side: /right_robotiq/-->
  <arg name="output_joints" default="/joint_states" />
  
  <node name    = "right_robotiq_republisher" 
        pkg     = "topic_tools" 
        type    = "relay" 
        ns      = "right_robotiq"
        args    = "$(arg input_joints)
                   $(arg output_joints)">
  </node>

  <arg name="output_joints2" default="/grasp_control/r_hand/joint_states" /><!-- Relaying to /grasp_control for OCS and Pick controllers-->
  
  <node name    = "right_grasp_control_republisher" 
        pkg     = "topic_tools" 
        type    = "relay" 
        ns      = "right_robotiq"
        args    = "$(arg input_joints)
                   $(arg output_joints2)">
  </node>
</launch>
